### DISCUSSION TITLE
Debugging and Relesse OK, but after published. the exe throws ButtonsListView is not efined yet. Please use ItemsSource instead.

### DISCUSSION THREAD

**Discussion by KunLee** [MahApps/MahApps.Metro#4420] [Q&A]
Hi guys,

I built a WPF application using MahApps.Metro. Debug/Release both running OK. But after publishing, running the exe throws error
'ButtonsListView is not defined yet. Please use ItemsSource instead.'
And 'The type initializer for System.Windows.Media.FontFamily threw and exception' and then the command window just stop there forever.

If I run the exe in debug or release folder directly, it is fine.

My prj file, I published it as self-contained

 <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net7.0-windows</TargetFramework>
    <Nullable>enable</Nullable>
    <UseWPF>true</UseWPF>
    <OutputType>Exe</OutputType>
    <PublishReadyToRun>true</PublishReadyToRun>
    <PublishSingleFile>true</PublishSingleFile>
    <RuntimeIdentifier>win-x64</RuntimeIdentifier>
  </PropertyGroup>

Any advice or suggestion would be highly appreciated.

**Comment by timunie** [comment]
are you using trimming? If yes, WPF doesn't really support it yet.

**Comment by KunLee** [comment]
Hi Tim,
Thanks for replying.
Nope. Not that but turned the ReadyToRun on.
Now I turned that one off and only keep SingleFile. It is okay.
Regards
Kun

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
